docs/user-guide/rug/tests.md

```diff
@@ -201,48 +358,220 @@ Given("a Java source file", (p: Project) => {
 });
 ```
 
-## Running Rugs
+There are also well-known steps for populating the project to be
+tested with the contents of the Rug project itself.  The first copies
+the entire contents of the Rug project archive, including the
+`.atomist` directory, into the project being used in the test.
+
+```gherkin
+Given the archive root
+```
+
+The second form copies everything *but* the `.atomist` directory into
+the project being used in the test.
+
+```gherkin
+Given archive non Atomist content
+```
+
+## Execution
+
+For each scenario there is typically one `#!gherkin When` step
+triggering a single action, sending an event or invoking a command
+handler, editor, or generator.
+
+### Triggering command handlers
+
+A command handler is typically triggered from within chat by someone
+sending the Atomist Bot a message with the command handler's intent.
+To simulate this action in the testing framework, you use the
+`#!typescript invokeHandler` method.
+
+```typescript
+import { CommandHandlerScenarioWorld, When } from "@atomist/rug/test/handler/Core";
+
+When("the TestCommandHandler is invoked", (w: CommandHandlerScenarioWorld) => {
+    const handler = w.commandHandler("TestCommandHandler");
+    w.invokeHandler(handler, {});
+});
+```
+
+The `#!typescript CommandHandlerScenarioWorld` interface used above
+inherits from `#!typescript HandlerScenarioWorld`, adding methods
+specific to command handlers.  Before invoking the command handler,
+you use the `#!typescript CommandHandlerScenarioWorld` object to
+lookup the handler by name using the `#!typescript commandHandler`
+method.  The name you provide is the same as the name used in the
+`#!typescript @CommandHandler` decorator.
+
+### Sending events
+
+To simulate an event occurring when testing an event handler, you
+first create the appropriate event object from the classes in the
+`cortex/stub` directory.  Then, you use the `sendEvent` method of
```

Comment:
Should we refer back to http://docs.atomist.com/user-guide/rug/path-expressions/#cortex-graph for a bit of context?